### PR TITLE
Wazuh to baseline

### PR DESCRIPTION
**Pull request by apdibbo**
various changes to add wazuh to the VM baseline

### CODE REVIEW

File: os_builders/roles/vm_baseline/tasks/wazuh.yml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        state: present
+
+    - name: Add Wazuh repo
+      ansible.builtin.apt_repository:
+        repo: "deb https://packages.wazuh.com/4.x/apt/ stable main"
+        state: present
+
+    - name: Install Wazuh agent
+      ansible.builtin.apt:
+        name: wazuh-agent=4.8.*
```
Comment (DavidFair) [inline os_builders/roles/vm_baseline/tasks/wazuh.yml:15]: Can we pin the major version of Wazuh and let me minor version float too? I'm assuming they use semver so 4.9 would be compatible with a 4.8 server?

File: os_builders/roles/vm_baseline/tasks/wazuh.yml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      dest: /var/ossec/etc/extra/01-ansible-baseline.json
+      content: "{{ wazuh_extra_json }}"
+
+  - name: Copy wazuh client enrichment script
+    copy:
+      src: "wazuh/enrich-wazuh-agent.py"
+      dest: "/var/ossec/bin/enrich-wazuh-agent.py"
+      owner: root
+      group: wazuh
+      mode: 0750
```
Comment (DavidFair) [inline os_builders/roles/vm_baseline/tasks/wazuh.yml:76]: Nit: we lock users out from reading this file even though the parent dir is read-only

Could we do something like: 0754 - so users can see exactly what is running without having the exec bit?

File: os_builders/roles/vm_baseline/tasks/wazuh.yml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+      group: root
+      mode: 0644
+
+  - name: Copy wazuh config template
+    copy:
+      src: "wazuh/ossec.template"
+      dest: "/var/ossec/etc/ossec.template"
+      owner: root
+      group: wazuh
+      mode: 0750
```
Comment (DavidFair) [inline os_builders/roles/vm_baseline/tasks/wazuh.yml:100]: Ditto on these perms ^

File: os_builders/roles/vm_baseline/tasks/wazuh.yml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+  - name: get service facts
+    ansible.builtin.service_facts:
+
+  - name: Stop wazuh-agent service
+    become: true
+    ansible.builtin.command:
+      cmd: "systemctl stop wazuh-agent.service"
+    ignore_errors: true
+
+  - name: Wazuh agent history
```
Comment (DavidFair) [inline os_builders/roles/vm_baseline/tasks/wazuh.yml:131]: - name: Clean Wazuh agent history

File: os_builders/roles/vm_baseline/tasks/wazuh.yml (hunk 1 above)
Comment (apdibbo) [reply os_builders/roles/vm_baseline/tasks/wazuh.yml:15]: Unfortunately the agent is not forwards compatible:
```
Compatibility between the Wazuh agent and the Wazuh manager is guaranteed when the Wazuh manager version is later than or equal to that of the Wazuh agent.
```

File: os_builders/roles/vm_baseline/tasks/wazuh.yml (hunk 4 above)
Comment (apdibbo) [reply os_builders/roles/vm_baseline/tasks/wazuh.yml]: Done

File: os_builders/roles/vm_baseline/tasks/wazuh.yml (hunk 3 above)
Comment (apdibbo) [reply os_builders/roles/vm_baseline/tasks/wazuh.yml]: Done

File: os_builders/roles/vm_baseline/tasks/wazuh.yml (hunk 2 above)
Comment (apdibbo) [reply os_builders/roles/vm_baseline/tasks/wazuh.yml]: Done

**Review by DavidFair — APPROVED** [review]

**Review by jacob-ward — APPROVED** [review]

**Pull request merged**